pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -88,6 +88,11 @@ def on_run_start(  # type: ignore[override]
         # add the previous `fetched` value to properly track `is_last_batch` with no prefetching
         data_fetcher.fetched += self.batch_progress.current.ready
 
+        data_fetcher._start_profiler = self.trainer.profiler.start
+        data_fetcher._stop_profiler = self.trainer.profiler.stop
+        suffix = "test_dataloader_next" if self.trainer.testing else "val_dataloader_next"
+        data_fetcher._profiler_action_name = f"[{self.__class__.__name__}].{suffix}"
```

Comment:
Let's avoid the `_profiler_action_name` attribute in the fetcher

```python
data_fetcher._start_profiler = self._on_before_fetch

def on_before_fetch(self):
    suffix = "test_dataloader_next" if self.trainer.testing else "val_dataloader_next"
    self.trainer.profiler.start(f"[{self.__class__.__name__}].{suffix}")
```


You could also use `functools.partial` but I would say this is nicer

Reply:
Didn't know about `functools.partial`, thanks for sharing!
I agree, this is approach is cleaner.